core.js

```diff
@@ -0,0 +1,685 @@
+/**
+ * Spacebrew Manager
+ * @module spacebrew/manager
+ */
+
+var Leaf = require('./leaf.js');
+var Route = require('./route.js');
+var Admin = require('./admin.js');
+
+/**
+ * Creates an instance of a Spacebrew Manager
+ * @constructor
+ */
+var Manager = function(){
+  /** the clients registered with this network */
+  this.clients = [];
+  /** the routes configured on this network */
+  this.routes = [];
+  /** the admins registered with this network */
+  this.admins = [];
+  /** the other managers in this network */
+  this.managers = [];
+};
+
+/**
+ * Adds a client (leaf node) to the server
+ * @param client {Leaf} The client object to add. 
+ * @returns {boolean} true iff the provided client is added. 
+ *   The client will not be added if it matches a client already registered.
+ * @throws Error if argument is not a Leaf instance
+ */
+Manager.prototype.addClient = function(client){
+  //make sure the client is the proper type
+  if (!(client instanceof Leaf)){
+    throw new TypeError('argument must be a Leaf instance');
+  }
+  //make sure there is not a duplicate client already registered
+  var clientUnique = (this.indexOfClient(client) === -1);
+  if (clientUnique){
+    //connect this client's publishers and subscribers
+    // to other endpoints based on the currently defined routes
+    this.connectClient(client);
+
+    //add the client to the current list of clients
+    this.clients.push(client);
+
+    //update the admins
+    for(var adminI = this.admins.length - 1; adminI >= 0; adminI--){
+      this.admins[adminI].sendAdd([client.toMap()], 
+                                  undefined, 
+                                  client.getConnections());
+    }
+  }
+  return clientUnique;
+};
+
+Manager.prototype.addAdmin = function(admin){
+  //make sure the admin is the proper type
+  if (!(admin instanceof Admin)){
+    throw new TypeError('argument must be an Admin instance');
+  }
+  //make sure there is not a duplicate admin already registered
+  var adminUnique = (this.indexOfAdmin(admin) === -1);
+  if (adminUnique){
+    //send current state to this admin
+    admin.sendAdd(this.getClients(), this.getRoutes(), this.getConnections());
+
+    //add the admin to the list of registered admins
+    this.admins.push(admin);
+  }
+  return adminUnique;
+};
+
+Manager.prototype.connectRoutePubClient = function(route, pubClient){
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Function 'anonymous' has a complexity of 8.](https://www.codacy.com/app/quinkennedy/spacebrew/file/2279533613/issues/source?bid=3184377&fileBranchId=3187330#l74)